elastic_transport/_node/_http_requests.py

```diff
@@ -169,7 +169,18 @@ def __init__(self, config: NodeConfig):
         )
         # Preload the HTTPConnectionPool so initialization issues
         # are raised here instead of in perform_request()
-        adapter.get_connection(self.base_url)  # type: ignore[no-untyped-call]
+        if hasattr(adapter, "get_connection_with_tls_context"):
+            request = requests.Request(method="GET", url=self.base_url)
+            prepared_request = self.session.prepare_request(request)
+            adapter.get_connection_with_tls_context(
+                prepared_request, verify=self.session.verify
+            )
+        else:
+            # elastic-transport is not vulnerable to CVE-2024-35195 because it uses
+            # requests.Session and an SSLContext without using the verify parameter.
+            # We should remove this branch when requiring requests 2.32 or later.
+            adapter.get_connection(self.base_url)
```

Comment:
Do we need to preserve the `type: ignore` here? And are there any typing considerations for the new call?

Reply:
We don't, the types were fixed in typeshed yesterday: https://github.com/python/typeshed/pull/12000.